network/wsNetwork.go

```diff
@@ -1144,6 +1205,20 @@ func (wn *WebsocketNetwork) ServeHTTP(response http.ResponseWriter, request *htt
 		challenge = wn.prioScheme.NewPrioChallenge()
 		responseHeader.Set(PriorityChallengeHeader, challenge)
 	}
+
+	var peerPublicKey crypto.PublicKey
+	var peerIDChallenge [32]byte
+	// if the caller supplied an identity challenge, inspect it and attach a response if it is valid and meant for us
+	idChalB64 := request.Header.Get(ProtocolConectionIdentityChallengeHeader)
+	if idChalB64 != "" {
```

Comment:
This deserves a 

` && wn.config.ConnectionDeduplicationName != ""`

In order to prevent participation in Identification if this host isn't set up for it.